packages/create-content-sdk-app/src/templates/nextjs/copilot-instructions.md

```diff
@@ -0,0 +1,158 @@
+- Enable strict null checks and strict function types
+
+### Naming Conventions
+- **Variables/Functions**: camelCase (`getUserData()`, `isLoading`, `currentUser`)
+- **Components**: PascalCase (`SitecoreComponent`, `PageLayout`, `ContentBlock`)
+- **Constants**: UPPER_SNAKE_CASE (`API_ENDPOINT`, `DEFAULT_TIMEOUT`)
+- **Directories**: kebab-case (`src/components`, `src/api-clients`)
+- **Types/Interfaces**: PascalCase (`ContentItem`, `LayoutProps`, `SitecoreConfig`)
+
+### Modular Layout
+```
+src/
+  components/          # UI components (React)
+  lib/                 # Configuration and utilities
+  pages/               # Next.js pages (or app/ for App Router)
+  assets/              # Static assets and styles
+  types/               # TypeScript type definitions
+  hooks/               # Custom React hooks
+```
+
+## Library Usage
+
+### @sitecore/content-sdk
+- Use `SitecoreClient` for content fetching
+- Implement proper error handling with try/catch blocks
+- Cache API responses using React Query or SWR
+- Handle content preview vs. published content scenarios
+
+```typescript
+import { SitecoreClient } from '@sitecore/content-sdk';
+
+const client = new SitecoreClient({
+  sitecoreApiHost: process.env.SITECORE_API_HOST,
+  sitecoreApiKey: process.env.SITECORE_API_KEY,
+  siteName: process.env.SITECORE_SITE_NAME
+});
+```
+
+### React Patterns
+- Use **Server Components** for data fetching and static content
+- Use **Client Components** for interactivity (use 'use client')
+- Implement proper error boundaries
+- Use React.memo for expensive components
+- Leverage useCallback and useMemo for performance optimization
+
+### Sitecore Field Components
+- Always use Sitecore field components: `<Text>`, `<RichText>`, `<Image>`
+- Validate field existence before rendering
+- Handle empty/null fields gracefully
+- Prefer Sitecore field components over manual rendering
+
+```typescript
+// Good: Using Sitecore field components
+<Text field={fields?.title} tag="h1" />
+<RichText field={fields?.content} />
+<Image field={fields?.backgroundImage} />
+
+// Avoid: Manual field value extraction unless necessary
+```
+
+## Example Patterns and Prompts
+
+### Component Development
+```typescript
+// Component props interface
+interface HeroProps {
+  fields: {
+    title: Field;
+    subtitle: Field;
+    backgroundImage: Field;
+  };
+}
+
+export default function Hero({ fields }: HeroProps) {
+  return (
+    <div>
+      <Text field={fields?.title} tag="h1" />
+      <Text field={fields?.subtitle} tag="p" />
+      <Image field={fields?.backgroundImage} />
+    </div>
+  );
+}
+```
+
+### Error Handling
+```typescript
+async function fetchContent(id: string): Promise<ContentItem> {
+  if (!id) {
+    throw new Error('Content ID is required');
+  }
+
+  try {
+    const response = await sitecoreClient.getItem(id);
+    return response.data;
+  } catch (error) {
+    throw new SitecoreFetchError(`Failed to fetch content ${id}`, error);
+  }
+}
```

Comment:
You can see one of the previous comments, this one is the same related to `getItem` and `contentId`